# HG changeset patch
# Parent  38d48db62539afe61d542c9d21e32d57d4b00a73
Eliminate startup error message:
JavaScript error: , line 0: Error: Type error for platformInfo value (Error processing arch: Invalid enumeration value "s390x") for runtime.getPlatformInfo.

Reported here: https://bugzilla.mozilla.org/show_bug.cgi?id=1554971

Uncertain if this is causing real problems or not. Also uncertain if the fix actually fixes anything.
No response from upstream yet.

diff --git a/toolkit/components/extensions/schemas/runtime.json b/toolkit/components/extensions/schemas/runtime.json
--- a/toolkit/components/extensions/schemas/runtime.json
+++ b/toolkit/components/extensions/schemas/runtime.json
@@ -59,17 +59,17 @@
         "type": "string",
         "allowedContexts": ["content", "devtools"],
         "description": "The operating system the browser is running on.",
         "enum": ["mac", "win", "android", "cros", "linux", "openbsd"]
       },
       {
         "id": "PlatformArch",
         "type": "string",
-        "enum": ["arm", "x86-32", "x86-64"],
+        "enum": ["arm", "x86-32", "x86-64", "s390x", "aarch64", "ppc64le"],
         "allowedContexts": ["content", "devtools"],
         "description": "The machine's processor architecture."
       },
       {
         "id": "PlatformInfo",
         "type": "object",
         "allowedContexts": ["content", "devtools"],
         "description": "An object containing information about the current platform.",
